enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Baremetal IPI Config Enhancement
+authors:
+  - "@sadasu"
+reviewers:
+  - "@deads2k"
+  - "@hardys"
+  - "@enxebre"
+  - "@squeed"
+  - "@abhinavdahiya"
+  - "@dhellmann"
+
+approvers:
+  - "@abhinavdahiya"
+  - "@enxebre"
+  - "@deads2k"
+
+creation-date: 2019-11-14
+last-updated: 2019-11-26
+status: not implemented
+see-also:
+  - 
+replaces:
+  - https://github.com/openshift/enhancements/pull/90
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [*] Enhancement is `implementable`
+- [*] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions
+
+1. There are no issues reading this new CR within operator scope from the
+Machine API Operator.
+
+2. Since this enhancement did not make it into 4.3 and the BareMetal IPI
+deployments are in Tech Preview in 4.2, the upgrade strategy should include
+upgrades from 4.2 to 4.3 to 4.4. (Assuming this enhancement makes it into 4.4.)
+
+## Summary
+
+The configuration required to bring up a Baremetal IPI (metal3) cluster needs
+to be availabe in a Config Resource. We had submitted an earlier proposal [1]
+to augment the BareMetalInfrastructureStatus CR with these config items. Based
+on feedback we recieved we are revising our proposal to create a new CR for
+these configuration items.
+
+This new CR needs to be accessed by just the Installer and the Machine API
+Operator (MAO) and hence does not need to be created with global scope.
+
+## Motivation
+
+The Baremetal IPI deployments are different from the other platform types currently
+being supported by OpenShift in that there is no underlying cloud controller
+that exists until provisioning of baremetal servers takes place first.
+
+The "metal3" pod deployed by the machine API operator (MAO) contains the BaremMetal
+Operator (BMO) and a provisioning service (Ironic) that are together responsible for
+PXE booting baremetal servers and enrolling them as BareMetal Hosts.
+
+This enhancement request proposes to add a new CR for configs required only for
+the provisioning service to PXE boot a baremetal server and make it available as a
+BareMetal host. The configurations in this baremetal provisioning CR, are seperate
+from the configuration in the BareMetalHost CR. This new CR will allow for setting
```

Comment:
This is all true, but based on the misunderstandings in #90 I'd perhaps add something like "this baremetal provisioning CR contains configuration data for the provisioning services, which are not values that should be configured via BareMetalHost objects by the end user, for example the subnet used for the dedicated provisioning network"

Reply:
Done.